### PR TITLE
HITL Band-Aid

### PR DESCRIPTION
**Pull request by shihaocao**
- Draft PR

---
This change is [Reviewable](https://reviewable.io/reviews/pathfinder-for-autonomous-navigation/flightsoftware/251)

### CODE REVIEW

File: ptest/state_session.py [hunk 1]
```diff
@@ -189,6 +189,20 @@ def _write_state_basic(self, fields, vals, timeout = None):
         for field in fields:
             returned_vals.append(self._wait_for_state(field, timeout))
 
+        if returned_vals[0] is None:
```
Comment (tanishqaggarwal) [inline ptest/state_session.py]: There's a feature in flight software (in QuakeManager) that can also return telemetry packets #157. Can you make sure that this new code doesn't break that feature?

File: src/fsw/FCCode/ADCSBoxController.cpp [hunk 2]
```diff
@@ -93,14 +93,29 @@ void ADCSBoxController::execute(){
...
+        bool disable_get = havt_cmd_disable_vector_fp[idx]->get();
+        if(disable_get){
+            temp_cmd_table.set(idx, disable_get);
+            send_cmd_table = true;
+        }
     }
-    adcs_system.set_havt_disable(temp_cmd_table);
+    
+    if(send_cmd_table)
+        adcs_system.set_havt_disable(temp_cmd_table);
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxController.cpp]: Can you add some comments for this logic? It's a little hard to follow at this point.

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 3]
```diff
@@ -129,7 +129,8 @@ void ADCSBoxMonitor::execute(){
     float gyr_temp = 0.0;
 
     //ask the driver to fill in values
-    adcs_is_functional.set(adcs_system.is_functional());
+    adcs_is_functional.set(adcs_system.i2c_ping());
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp]: Did you switch this to get rid of the persistent err_count behavior?

File: src/fsw/FCCode/DebugTask.cpp [hunk 4]
```diff
@@ -16,8 +16,8 @@ DebugTask::DebugTask(StateFieldRegistry &registry, unsigned int offset)
 
 void DebugTask::execute() {
 #ifdef FUNCTIONAL_TEST
-  start_cycle_f.set(false);
-  while (!start_cycle_f.get())
+  // start_cycle_f.set(false);
+  // while (!start_cycle_f.get())
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/DebugTask.cpp]: Can you put this back in

File: ptest/state_session.py [hunk 5]
```diff
@@ -189,6 +189,20 @@ def _write_state_basic(self, fields, vals, timeout = None):
...
+            return False
+
+        returned_vals = returned_vals[0].split(",")
+        returned_vals = [x for x in returned_vals if x is not ""]
+        
+        if (returned_vals[0].replace('.','')).isnumeric():
```
Comment (kkrol27) [inline ptest/state_session.py]: Negative numbers here?

File: src/adcs/havt_devices.hpp [hunk 6]
```diff
@@ -36,6 +36,7 @@ enum Index : unsigned char {
   SSA_ADC1,
   SSA_ADC2,
   SSA_ADC3,
+
```
Comment (kkrol27) [inline src/adcs/havt_devices.hpp]: This extra line intentional?

File: src/adcs/rwa.cpp [hunk 7]
```diff
@@ -196,6 +196,8 @@ void actuate(unsigned char rwa_mode, lin::Vector3f rwa_cmd) {
       // Set wheel configurations
       for (unsigned int i = 0; i < 3; i++) {
         if (wheels[i].is_functional()) {
+          LOG_INFO_header
+          LOG_INFO_println("wheel act: " + String(rwa_cmd(i)))
```
Comment (kkrol27) [inline src/adcs/rwa.cpp]: Wondering here if we really want this at the `INFO` logging level. I don't see why we wouldn't just set the `rwa` module itself to the `TRACE` level and let everything else be?

File: src/adcs/state_controller.cpp [hunk 8]
```diff
@@ -17,6 +17,10 @@
   #define LOG_LEVEL UMB_LOG_LEVEL
 #endif
 
+#undef LOG_LEVEL
+#define LOG_LEVEL 0
+// ^ The above is a fix to prevent serial debug output during the i2c interrupt.
```
Comment (kkrol27) [inline src/adcs/state_controller.cpp]: This is fine for now. In the future though we should probably just zero out the logging level in the config file and call it good.

File: src/adcs/havt_devices.hpp (hunk 6 above)
Comment (shihaocao) [reply src/adcs/havt_devices.hpp]: Yep just helps us count devices

File: src/fsw/FCCode/ADCSBoxMonitor.cpp (hunk 3 above)
Comment (shihaocao) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp]: is_functional() is tied to the error_count field which is latching. So I changed it to i2c_ping() so that it would always reflect the true functionality of the ADCSC in the current moment (which I think is more desirable).

File: ptest/state_session.py [hunk 9]
```diff
@@ -195,7 +195,7 @@ def _write_state_basic(self, fields, vals, timeout = None):
         returned_vals = returned_vals[0].split(",")
         returned_vals = [x for x in returned_vals if x is not ""]
         
-        if (returned_vals[0].replace('.','')).isnumeric():
+        if (returned_vals[0].replace('.','').replace('-','')).isnumeric():
```
Comment (tanishqaggarwal) [inline ptest/state_session.py]: Why was this needed? Are negative numbers not numeric?

File: src/adcs/state_controller.cpp (hunk 8 above)
Comment (shihaocao) [reply src/adcs/state_controller.cpp]: Should be cleaner now, realized can use #undef to fix compilation errors for log level selection

File: src/adcs/rwa.cpp (hunk 7 above)
Comment (shihaocao) [reply src/adcs/rwa.cpp]: Oops left that in accidentally

File: ptest/state_session.py (hunk 5 above)
Comment (shihaocao) [reply ptest/state_session.py]: Fixed

File: ptest/state_session.py (hunk 1 above)
Comment (shihaocao) [reply ptest/state_session.py]: Verified over PM that seems like still working

File: ptest/state_session.py (hunk 9 above)
Comment (shihaocao) [reply ptest/state_session.py]: isnumeric() is true iff all the string characters contain one of 1234567890

File: ptest/cases/spin_motors_case.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# SpinMotorsCase. Gets satellite read to spin motors.
+from .base import SingleSatOnlyCase
+
+class SpinMotorsCase(SingleSatOnlyCase):
+    def setup_case_singlesat(self):
+        self.sim.flight_controller.write_state("pan.state", 11) # Mission State = Manual
+        self.sim.flight_controller.write_state("adcs.state", 5) # ADCS State = Manual
+        self.sim.flight_controller.write_state("adcs_cmd.rwa_mode", 1) # Speed Control
+        self.sim.flight_controller.write_state("adcs_cmd.rwa_speed_cmd", "0, 0, 0") # 0 speed to begin with
```
Comment (kkrol27) [inline ptest/cases/spin_motors_case.py]: It'd be cool if we could potentially expand the `write_state` function to take `*args` instead of just one `arg`. Separate issue though for sure.

File: ptest/cases/spin_motors_case.py [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+        self.sim.flight_controller.write_state("dcdc.ADCSMotor_cmd", "true")
```
Comment (kkrol27) [inline ptest/cases/spin_motors_case.py]: Also potentially support for booleans?

File: src/adcs/main.cpp [hunk 12]
```diff
@@ -216,6 +216,7 @@ void update_ssa() {
 
 #if LOG_LEVEL >= LOG_LEVEL_INFO
 static unsigned long cycles = 0;
+static unsigned long last_info_time = millis();
```
Comment (kkrol27) [inline src/adcs/main.cpp]: Working off time is definitely a better move!

Comment (shihaocao) [reply src/adcs/main.cpp]: :)

File: ptest/cases/spin_motors_case.py (hunk 11 above)
Comment (shihaocao) [reply ptest/cases/spin_motors_case.py]: yeah true, that would be a good upgrade.
